### PR TITLE
Add Managed Instance Group template

### PR DESCRIPTION
**Pull request by sourced-pavel**

### CODE REVIEW

File: community/cloud-foundation/templates/managed_instance_group/managed_instance_group.py.schema [hunk 1]
```diff
@@ -227,23 +227,23 @@ properties:
...
         description: |
-          Configuration parameters of autoscaling based on load balancer.
+          The autoscaling configuration prameters based on the load balancer.
         properties:
           utilizationTarget:
             type: number
             minimum: 0
             maximum: 1
-            description: Fraction of backend capacity utilization.
+            description: The ??? required? minimum? maximum ??? fraction ??? in percents? or... ??? of back-end capacity utilization.
```
Comment (sourced-pavel) [inline community/cloud-foundation/templates/managed_instance_group/managed_instance_group.py.schema]: @alexander-levitski Not sure that I understand your question. This is the [full description](https://cloud.google.com/compute/docs/reference/rest/v1/autoscalers) from Google:
`Fraction of backend capacity utilization (set in HTTP(s) load balancing configuration) that autoscaler should maintain. Must be a positive float value. If not defined, the default is 0.8.`
The meaning of the value is either CPU (`1` being `100%` load) or Requests Per Second, whose maximum value is set manually when another resource type - backend service - is created

File: community/cloud-foundation/templates/managed_instance_group/managed_instance_group.py [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+
+REGIONAL_LOCAL_AUTOSCALER_TYPES = {
+    True: 'compute.v1.regionAutoscaler',
+    False: 'compute.v1.autoscaler'
+}
+REGIONAL_LOCAL_IGM_TYPES = {
+    True: 'compute.v1.regionInstanceGroupManager',
+    False: 'compute.v1.instanceGroupManager'
+}
+
```
[the remark this answers is not in the document]
Comment (sourced-pavel) [reply community/cloud-foundation/templates/managed_instance_group/managed_instance_group.py]: I actually agree with you on that. There was something wrong with our priorities list, so MIG template was put above autoscaler and instance_template templates. I just finished autoscaler and my intention was to reuse it from MIG once autoscaler gets approved.
It makes sense to implement instance_template now, and only then get back to MIG.
Thanks for bringing that up.

[the remark this answers is not in the document]
Comment (sourced-praveenc) [reply community/cloud-foundation/templates/managed_instance_group/managed_instance_group.py]: Thanks. I'll resolve this comment for now

**Comment by sourced-pavel** [comment]
Closing for now. Depends on `autoscaler` and `instance_template` templates being merged into google's cloud-foundation branch

**Pull request closed without merging**